### PR TITLE
[TFLite] Clarify Interpreter input detail ordering

### LINKED ISSUE #123996: TFLiteConverter produces a wrong value from a single tf.nn.top_k call, and can also get the index itself wrong
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes. I reproduced this with:

```text
tf-nightly 2.22.0-dev20260713
```

(also reproduces identically on `tf-nightly 2.22.0-dev20260531`). The `tflite value` output is exactly `44.0` on both nightly builds too, identical to what I saw on TF 2.20.0 — not just "still wrong", but bit-identical wrong output across all three versions tested.

### Source

source

### TensorFlow version

2.20.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu (x86_64)

### Python version

3.9

### Current behavior?

A model containing two `tf.nn.top_k` calls (the first one's `indices` output discarded in Python, the second one's `values`/`indices` both returned) converts without error, but the converted model's `values` output does not match eager execution, and in some variants the `indices` output is wrong too.

I cross-checked with 4 variants that only differ in the activation function applied between the two `top_k` calls (`gelu`, `leaky_relu`, `sigmoid`, `cos`), matching TFLite's two outputs to eager's `(values, indices)` by dtype (not position), to rule out an output-ordering artifact:

| variant | eager index | tflite index | eager value | tflite value |
|---|---|---|---|---|
| gelu | 1 | 1 | 47.97 | 44.0 (wrong) |
| leaky_relu | 1 | 1 | 48.0 | 44.0 (wrong, identical to the row above) |
| sigmoid | 1 | 1 | 13.51 | 5.996 (wrong) |
| cos | 0 | 1 (wrong) | -10.20 | 2.725 (wrong) |

The `cos` variant shows the index itself can be wrong too, not just the value — TFLite's index output is `1` in all four variants regardless of input, while the true argmax in the `cos` variant is at position `0`.

Expected: the converted model's `top_k` output should match eager execution.

### Standalone code to reproduce the issue

```python
import tensorflow as tf

class Model(tf.keras.Model):
    def __init__(self):
        super(Model, self).__init__()
        self.m1 = tf.Variable([1, 2])  # declared but unused in call()

    def call(self, x1, x2):
        x3, _ = tf.nn.top_k(x1, 2)      # first top_k call, indices discarded
        v2_0 = tf.nn.gelu(x3)
        v3_0 = tf.reshape(x2, (2, 2))
        x4 = tf.matmul(v2_0, v3_0)
        return tf.nn.top_k(x4, 1)       # second top_k call, both outputs returned

m = Model()
x1 = tf.constant([1., 2, 3, 4], shape=[1, 4])
x2 = tf.constant([5., 6, 7, 8], shape=[1, 4])
input_data = [x1, x2]
print('expected (eager):', m(*input_data))

class Model(tf.keras.Model):
    def __init__(self):
        super(Model, self).__init__()
        self.m1 = tf.Variable([1, 2])

    @tf.function(input_signature=[tf.TensorSpec(shape=x.shape, dtype=x.dtype) for x in input_data])
    def call(self, x1, x2):
        x3, _ = tf.nn.top_k(x1, 2)
        v2_0 = tf.nn.gelu(x3)
        v3_0 = tf.reshape(x2, (2, 2))
        x4 = tf.matmul(v2_0, v3_0)
        return tf.nn.top_k(x4, 1)

m2 = Model()
converter = tf.lite.TFLiteConverter.from_keras_model(m2)
tflite_model = converter.convert()
interpreter = tf.lite.Interpreter(model_content=tflite_model)
interpreter.allocate_tensors()
ins = interpreter.get_input_details()
outs = interpreter.get_output_details()
for i, d in enumerate(input_data):
    interpreter.set_tensor(ins[i]['index'], d.numpy())
interpreter.invoke()
for o in outs:
    print('actual (tflite):', o['name'], o['dtype'], interpreter.get_tensor(o['index']))
```

### Relevant log output

```text
expected (eager): TopKV2(values=47.966843, indices=1)
actual (tflite):  int32 output   = [[1]]     # index: correct here, but not in the `cos` variant (see above)
                   float32 output = [[44.]]   # value: wrong, doesn't match any element of x4
```

### PR DESCRIPTION
**Pull request by GodlyDonuts**
## Summary
- document that `get_input_details()` order may differ from the original TensorFlow function argument order
- direct signature-bearing models to `get_signature_runner()` for name-based input binding

The reported TopK values are correct when `x1` and `x2` are bound by signature name. Positional zipping reverses the two inputs in this model because the interpreter input list is `x2`, then `x1`.

## Testing
- verified the issue reproducer with positional and signature-based input binding
- Python syntax check for `interpreter.py`

Closes #123996

### CODE REVIEW

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
While this PR correctly addresses input ordering in `get_input_details()`, `get_output_details()` should also receive a similar disclaimer. Output tensor ordering may also differ from signature return order or original function return positions. Could you please add this clarification to `get_output_details()` as well?

**Comment by GodlyDonuts** [comment]
> While this PR correctly addresses input ordering in `get_input_details()`, `get_output_details()` should also receive a similar disclaimer. Output tensor ordering may also differ from signature return order or original function return positions. Could you please add this clarification to `get_output_details()` as well?

Addressed in `4c4c1f6` by adding the corresponding ordering clarification to `get_output_details()` as well. Thanks for catching this!

**Review by dmiltr3 — APPROVED** [review]
Thank you for adding the corresponding tensor ordering disclaimers to `get_output_details()` in addition to `get_input_details()`, and for pointing models with signatures toward `get_signature_runner()`. The documentation is clear, accurate, and completely resolves GitHub issue #123996. Looks great to merge.
